refactor: sales invoice integration with pos

diptanilsaha opened the pull request:
Changes Include:

- Replaced Check field with Select Field on POS Settings, allowing users to select Invoice DocType, which will be used in POS to create Invoice
		Before: ![image](https://github.com/user-attachments/assets/0530ced2-b748-4eed-a922-4f6850f5b92b)
		After: ![image](https://github.com/user-attachments/assets/2a5173de-eda8-4361-a5b5-397bebc58c06)

- Refactored as suggested in https://github.com/frappe/erpnext/pull/46485#pullrequestreview-2753190075
- Replaced the `get_doc` method in `get_invoices` with SQL using Query Builder.
- Rearranged fields and removed Sales Summary in POS Closing Entry.
- Refactored existing test cases and added tests for the Sales Invoice Integration with POS.

diptanilsaha committed:
fix: invoice doctype selection in accounts settings

diptanilsaha committed:
test: change in accounts settings on sales invoice

diptanilsaha committed:
test: refactored pos_invoice_merge_log tests

diptanilsaha committed:
test: pos closing entry and pos invoice

diptanilsaha committed:
fix: closing voucher details style

diptanilsaha committed:
refactor: renamed fields and removed repeated methods

diptanilsaha committed:
fix: patch to rename pos closing entry fields

diptanilsaha committed:
refactor: replaced get_doc with sql query

diptanilsaha committed:
fix: restrict cancelling sales invoice on cancellation of pos closing entry

diptanilsaha committed:
fix: removed payment reconciliation summary field and rearranged total section fields

diptanilsaha committed:
refactor: set_posting_date_and_time

diptanilsaha committed:
test: create_sales_invoice added args for is_created_using_pos

diptanilsaha committed:
test: added test for sales invoice creation during pos invoice mode

diptanilsaha committed:
test: added test for pos invoice creation during sales invoice mode

nabinhait commented on erpnext/accounts/doctype/accounts_settings/accounts_settings.json:
```diff
@@ -614,6 +607,14 @@
+   "description": "For high-volume transactions, it is advisable to use POS Invoice.",
+   "fieldname": "invoice_doctype_in_pos",
+   "fieldtype": "Select",
+   "label": "Invoice DocType",
```
Invoice Type Created via POS Screen

nabinhait replied:
Move to POS Settings

nabinhait commented on erpnext/accounts/doctype/accounts_settings/accounts_settings.json:
```diff
@@ -614,6 +607,14 @@
+  {
+   "default": "Sales Invoice",
+   "description": "For high-volume transactions, it is advisable to use POS Invoice.",
+   "fieldname": "invoice_doctype_in_pos",
```
similar with field label

nabinhait commented on erpnext/accounts/doctype/accounts_settings/accounts_settings.py:
```diff
@@ -145,14 +145,14 @@ def validate_and_sync_auto_reconcile_config(self):
 				frappe.throw(_("Queue Size should be between 5 and 100"))
 
-	def validate_invoice_mode_switch_in_pos(self):
+	def validate_invoice_type_switch_in_pos(self):
```
```suggestion
	def validate_invoice_type_in_pos(self):
```

nabinhait commented on erpnext/accounts/doctype/accounts_settings/accounts_settings.py:
```diff
@@ -145,14 +145,14 @@ def validate_and_sync_auto_reconcile_config(self):
+					frappe.bold(_("Invoice Type"))
 				),
-				title=_("Switch Invoice Mode Error"),
+				title=_("Switch Invoice DocType Error"),
```
```suggestion
				title=_("Switch Invoice Document Type Error"),
```

nabinhait commented on erpnext/accounts/doctype/pos_closing_entry/pos_closing_entry.json:
```diff
@@ -20,18 +20,19 @@
+  "sales_invoices",
   "section_break_11",
   "payment_reconciliation",
+  "taxes_and_charges_section",
+  "taxes",
   "section_break_13",
-  "grand_total",
-  "net_total",
-  "total_quantity",
   "column_break_16",
-  "taxes",
+  "total_quantity",
+  "column_break_ywgl",
+  "net_total",
+  "total_taxes_and_charges",
+  "grand_total",
```
Reorganize total fields and put payments table after totals

nabinhait commented on erpnext/accounts/doctype/pos_closing_entry/test_pos_closing_entry.py:
```diff
@@ -293,45 +304,208 @@ def test_merging_into_sales_invoice_for_batched_item(self):
+			self.assertIn(pos_inv1.name, [d.pos_invoice for d in pcv_doc1.pos_invoices])
+			self.assertEqual(pcv_doc1.grand_total, 300)
+
+		with self.change_settings("Accounts Settings", {"invoice_doctype_in_pos": "Sales Invoice"}):
```
put it in separate function

nabinhait commented on erpnext/accounts/doctype/accounts_settings/accounts_settings.json:
```diff
@@ -614,6 +607,14 @@
+  },
+  {
+   "default": "Sales Invoice",
+   "description": "For high-volume transactions, it is advisable to use POS Invoice.",
```
```suggestion
   "description": "The system will create a Sales Invoice or a POS Invoice from the POS interface based on this setting. For high-volume transactions, it is recommended to use POS Invoice.",
```

diptanilsaha committed:
fix: moved invoice type selection in pos settings

diptanilsaha committed:
fix: pos additional fields label

diptanilsaha committed:
refactor: pos closing entry

rearranged fields, removed rate field from taxes field, fetching payments and taxes details

diptanilsaha committed:
test: moved invoice creation in functions

diptanilsaha committed:
refactor: using as_dict=1

diptanilsaha committed:
fix: wrong table chosen in query

diptanilsaha committed:
fix: variable rename

diptanilsaha committed:
test: fixed failing tests

diptanilsaha committed:
test: fixed pos_closing_entry tests
